src/frontend/js/components/DashboardCard/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Icon } from 'components/Icon';
+import { FC, ReactNode, useRef, useState } from 'react';
+
+const DASHBOARD_CARD_EXPAND_DURATION = 100;
+
+interface DashboardCardProps {
+  header: ReactNode | string;
+  footer?: ReactNode;
+}
+
+export const DashboardCard: FC<DashboardCardProps> = (props) => {
+  const [opened, setOpened] = useState(true);
+  const expandableRef = useRef<HTMLDivElement>(null);
+  const [wrapperHeight, setWrapperHeight] = useState('auto');
+
+  const toggle = () => {
+    if (opened) {
+      setOpened(false);
+      // We need to set hard height first in order to trigger the transition.
+      setWrapperHeight(expandableRef.current!.getBoundingClientRect().height + 'px');
+      setTimeout(() => {
+        setWrapperHeight('0');
+      });
+    } else {
+      setOpened(true);
+      setWrapperHeight(expandableRef.current!.getBoundingClientRect().height + 'px');
+      setTimeout(() => {
+        // After the animation is completed, set height to auto to prevent hidden overflow
+        // if the content changes afterwards.
+        setWrapperHeight('auto');
+      }, DASHBOARD_CARD_EXPAND_DURATION);
+    }
+  };
+
+  const onHeaderClick = () => {
+    if (!opened) {
+      toggle();
+    }
+  };
+
+  const classes = ['dashboard-card'];
+  if (opened) {
+    classes.push('dashboard-card--opened');
+  } else {
+    classes.push('dashboard-card--closed');
+  }
+
+  return (
+    <div className={classes.join(' ')}>
+      <header className="dashboard-card__header" onClick={onHeaderClick}>
```

Comment:
:wave: 

For people using keyboard to navigate and interact on the page, you should avoid setting click event listeners on non-interactive HTML elements. Because with this implemented like this, I can't click on the header when using only a keyboard. It's not focusable, not in the tab order, not actionable via the enter/space keys, etc.

99% of the time, using a `button` element that you would style as required is the solution. I'd also note one good practice in a component-world like React would be to not forget to set the `type="button"` attribute on the button to prevent it from sending a potential form the component is included into.

Reply:
Note that, if the toggle is doable via a `button` wrapping the `Icon` inside this header, with the `Icon` having a `title` attribute (so that screen reader users get to know what the button does), there is no problem having a mouse-only event on the header then, as a bonus for mouse/touch users.

But as is, the header _and_ the icon are not keyboard-accessible.

Reply:
Thank you @Leimi for your feedback ! I will implements these changes for the final review, very instructive 🙏